### PR TITLE
Lendings whitelist filter

### PR DESCRIPTION
**Pull request by dudendy**

### CODE REVIEW

**Review by Vladislav94 — APPROVED** [review]

File: src/components/Filters/GotchiFilter.js [hunk 1]
```diff
@@ -9,75 +9,91 @@ import gotchiverseUtils from 'utils/gotchiverseUtils';
...
 
     useEffect(() => {
         if (!dataLoading) {
-            filterGotchis(guildsFilter);
+            handleQueryParams();
+            filterData();
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [dataLoading]);
+    }, [guildsFilter, whitelistFilter, dataLoading])
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:39]: semicolon missed

File: src/components/Filters/GotchiFilter.js [hunk 2]
```diff
@@ -9,75 +9,91 @@ import gotchiverseUtils from 'utils/gotchiverseUtils';
...
+        if (params.whitelist) {
+            setWhitelistFilter(params.whitelist);
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
     useEffect(() => {
         if (!dataLoading) {
-            filterGotchis(guildsFilter);
+            handleQueryParams();
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:35]: please pass `guildsFilter` and `whitelistFilter` as arguments to `handleQueryParams` and ` filterData` for clarity

File: src/components/Filters/GotchiFilter.js [hunk 3]
```diff
@@ -9,75 +9,91 @@ import gotchiverseUtils from 'utils/gotchiverseUtils';
...
 
-        if (filters.length > 0) {
-            filteredGotchis = gotchis.filter(gotchi =>
-                filters.some((filter) => filter === gotchi.guild)
-            );
-        } else {
-            filteredGotchis = gotchis;
-        }
+    const handleQueryParams = () => {
+        params.guild = guildsFilter.map((filter) => filter)
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:50]: missed semicolon

File: src/components/Filters/GotchiFilter.js [hunk 4]
```diff
@@ -9,75 +9,91 @@ import gotchiverseUtils from 'utils/gotchiverseUtils';
...
 
-    const handleQueryParams = (filters) => {
-        if (filters.length > 0) {
-            history.push({
-                path: location.pathname,
-                search: `?guild=${filters.map((filter) => filter)}`
-            });
+    const filterData = () => {
+        let filtered;
+        const isGuildsFilter = (guildsFilter.length > 0) === true;
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:61]: `=== true` is redundant

File: src/utils/commonUtils.js [hunk 5]
```diff
@@ -54,6 +54,10 @@ export default {
         return `${sellerAddress.substring(0, 4)}...${sellerAddress.substring(sellerAddress.length - 4, sellerAddress.length)}`;
     },
 
+    primitiveSort(array, sortDir) {
```
Comment (dmutrokot) [inline src/utils/commonUtils.js:57]: I would consider to rename function to `sortByDirection`, naming `primitiveSort` is not very informative)

File: src/pages/Lend/Lend.js [hunk 6]
```diff
@@ -85,14 +86,21 @@ export default function Lend() {
 
         thegraphApi.getLendings().then((response) => {
             if (mounted) {
-                const withGuilds = response.map((listing) => {
+                const whitelistData = [];
+                const mappedData = response.map((listing) => {
```
Comment (dmutrokot) [inline src/pages/Lend/Lend.js:90]: please use `forEach` as discussed

**Review by dmutrokot — CHANGES_REQUESTED** [review]

**Review by dmutrokot — APPROVED** [review]

**Pull request merged**